### PR TITLE
Update create, get, list, update to use new model; bump version

### PR DESCRIPTION
**Pull request by tkyi**
Use new pipeline model (https://github.com/screwdriver-cd/models/pull/4)

### CODE REVIEW

File: lib/create.js [hunk 1]
```diff
@@ -17,66 +15,28 @@ module.exports = (datastore) => ({
...
-                table: 'pipelines',
-                params: {
-                    id: pipelineId
-                }
-            };
+            const Pipeline = new Model.Pipeline(datastore);
 
             /* eslint-disable consistent-return */
-            datastore.get(config, (error, data) => {
+            Pipeline.get(request.payload, (error, data) => {
```
Comment (nkatzman) [inline lib/create.js]: Pipeline.get just takes an `id` https://github.com/screwdriver-cd/models/blob/master/lib/pipelinemodel.js#L94

File: lib/update.js [hunk 2]
```diff
@@ -26,7 +28,7 @@ module.exports = (datastore) => ({
                 }
             };
 
-            datastore.update(config, (err, response) => {
+            Pipeline.update(config, (err, response) => {
```
Comment (nkatzman) [inline lib/update.js]: Does Pipline.update return the full object that was merged?

Did we agree that we should be returning that back to the user?

File: lib/create.js (hunk 1 above)
Comment (tkyi) [inline lib/create.js]: Good catch! fixed

File: lib/update.js (hunk 2 above)
Comment (tkyi) [inline lib/update.js]: I think it makes sense to return the entire object back to the user.
But would it make more sense just to call a GET after successfully updating?

Comment (nkatzman) [inline lib/update.js]: I think update might return those values

Comment (nkatzman) [inline lib/update.js]: https://github.com/screwdriver-cd/screwdriver-datastore-imdb/blob/master/index.js#L100

Comment (d2lam) [inline lib/update.js]: ^^^  yup

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "screwdriver-plugin-pipelines",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (FenrirUnbound) [inline package.json]: Did the API change?

Comment (d2lam) [inline package.json]: Yeah don't need to bump version here.

File: lib/create.js [hunk 4]
```diff
@@ -17,66 +15,28 @@ module.exports = (datastore) => ({
...
-                    id: pipelineId
-                }
-            };
+            const Pipeline = new Model.Pipeline(datastore);
 
             /* eslint-disable consistent-return */
-            datastore.get(config, (error, data) => {
+            Pipeline.get(request.params.id, (error, data) => {
+                if (error) {
+                    return reply(error);
```
Comment (d2lam) [inline lib/create.js]: Let's wrap this with boom

File: lib/update.js (hunk 2 above)
Comment (tkyi) [inline lib/update.js]: @nkatzman updated

**Comment by FenrirUnbound** [comment]
:+1: 

Looks fine to me

8epc6cz

**Comment by d2lam** [comment]
👍

**Pull request merged**